sdk/src/resource_store.rs

```diff
@@ -170,6 +181,43 @@ impl ResourceStore {
         Ok(ResourceRef::new(format, id))
     }
 
+    /// Adds a resource from a uri, generating a resource ref
+    ///
+    /// The generated identifier may be different from the key
+    pub(crate) fn add_uri<R>(
+        &mut self,
+        uri: &str,
+        format: &str,
+        value: R,
+    ) -> crate::Result<ResourceRef>
+    where
+        R: Into<Vec<u8>>,
+    {
+        #[cfg(feature = "file_io")]
+        let mut id = uri.to_string();
+        #[cfg(not(feature = "file_io"))]
+        let id = uri.to_string();
+
+        // if it isn't jumbf, assume it's an external uri and use it as is
+        if id.starts_with("self#jumbf=") {
+            #[cfg(feature = "file_io")]
+            if self.base_path.is_some() {
+                // convert to a file path always including the manifest label
+                id = id.replace("self#jumbf=", "");
+                if id.starts_with("/c2pa/") {
+                    id = id.replace("/c2pa/", "");
```

Comment:
I didn't know about replacen, thanks for that. It isn't needed in this case since I'm bracketing it with slashes on both sides and that is defined as happening only once but still a good precaution.